docs/source/traits_user_manual/notification.rst

```diff
@@ -9,779 +9,622 @@ possible for trait attributes. This functionality lets you write programs
+
+.. rubric:: Examples of Traits DSL
+
+.. list-table::
+   :widths: 15 25
+   :header-rows: 1
+
+   * - Example
+     - Meaning
+   * - ``"foo.+updated"``
+     - Matches any trait on the *foo* that has a metadata attribute named
+       *updated*. Changes on those traits and changes on *foo* will trigger
+       notifications.
+   * - ``"foo,bar"``
+     - Matches trait named *foo* or *bar* on the current object. Changes on
+       *foo* or *bar* will trigger notifications.
+   * - ``"foo:[bar,baz]"``
+     - Matches *foo.bar* or *foo.baz* on the current object. Changes on
+       *foo.bar* or *foo.baz* will trigger notifications, but changes on *foo*
+       will not trigger notifications.
+   * - ``"container.items.value"``
+     - If *container* is a list or dict or set, matches the *value* trait on an
+       object that is an item of the container. If *container* is an instance
+       of *HasTraits*, matches attribute *container.items.value* on the current
+       object. Changes to any of these will trigger notifications.
+   * - ``"container:items:value"``
+     - If *container* is a list or dict or set, matches the *value* trait on an
+       object that is an item of the container. If *container* is an instance
+       of *HasTraits*, matches attribute *container.items.value* on the current
+       object. Only changes to *value* will trigger notifications, assignments
+       or mutations on *container* will not trigger notifications.
+
+.. _observe-expression-object:
+
+Expressions as objects
+``````````````````````
 
-      def update_cost(self):
-        self.cost = self.part1.cost + self.part2.cost
+|ObserverExpression| supports all the use cases supported by the Traits Mini Language
+and beyond. Users can compose these objects programmatically and supply
+them to |@observe|. These objects are typically constructed using the following
+functions from the |observation.api| module.
+
+.. list-table::
+   :widths: 15 25
+   :header-rows: 1
+
+   * - Function
+     - Purpose
+   * - |trait|
+     - For observing a specific named trait.
+   * - |metadata|
+     - For observing multiple traits with a specific metadata. Support further
+       filtering on the metadata value.
+   * - |dict_items|
+     - For observing items in a dict.
+   * - |list_items|
+     - For observing items in a list.
+   * - |set_items|
+     - For observing items in a set.
+   * - |match|
+     - For observing traits satisfying a user-defined filter.
+   * - |parse|
+     - For parsing a string written in Traits domain specific language into an
+       expression object.
+
+Users can combine complex expressions using |ObserverExpression.then| or
+Python's bitwise-or (`|`) operation.
+
+Most of these functions will have a **notify** argument for setting whether
+notifications should be fired for changes.
+
+.. rubric:: Example expressions
+
+.. list-table::
+   :widths: 15 25
+   :header-rows: 1
+
+   * - Expression
+     - Meaning
+   * - *trait("attr1")*
```

Comment:
Can we add backticks around these, so that the rendered expressions can be copy-pasted? (Currently they're appearing with code-unfriendly smart quotes.)

Reply:
They will look rather long and squash the "meaning" column into a small column.
I could instead break this out into bulletins instead of trying to fit them in a table.
Happy to do this in a separate PR so not to block the release process.

Reply:
That could work. Right now, the rendered code can't be copy-and-pasted because of the smart quotes.

Reply:
SGTM

Reply:
I thought it was going to take me a while to get the formatting right. It did not take so long at the end: See #1140